### DISCUSSION TITLE
How to add custom CSS only to specific elements?

### DISCUSSION THREAD

**Discussion by Anindya088** [zauberzeug/nicegui#1183] [Q&A] [answered]
### Question

Hi All,

I am trying to do some customizations for the elements. So let's say I am using 4 input boxes, and for one of them I want to shift the 'clearable' icon a bit to the left, so a bit of right padding.

In one of my ealier questions, I already got the answer on how to customize specific items by "inspect element". In fact this point was also mentioned about how to handle multiple cases by creating a custom class. But recently I tried applying that, and I can't seem to figure it out. 

So what I have is this below:
```python
ui.add_head_html("""
<style>
q-field__focusable-action {
    padding-right: 20px;
}
</style>
""")

input1 = ui.input('input 1').props('clearable')
input2 = ui.input('input 2').props('clearable')
input3 = ui.input('input 3').props('clearable')
input4 = ui.input('input 4').props('clearable')
```

That name ```q-field__focusable-action```, I got by inspect element on the webpage, and if I try this it works fine for all 4. But my challenge is how do I make it work only for 1 of the input boxes? 

What I understood is to convert the CSS code into a custom class like 
```python
.my-button.q-field__focusable-action {
```
and then apply as a class on the button like
```python
input4 = ui.input('input 4').props('clearable').classes('my-button')
```

But this seems not to work, I think I misunderstood the original solution or doing something wrong here. Can you please let me know what is the correct way to do this?

P.S. I think the other question is already resolved, that's why creating a new one, but if it is more appropriate to ask it as a follow up in that thread, please let me know, then I will do so and delete this one.

Thanks,
Anindya

**Comment by falkoschindler** [comment, marked as answer]
Hi @Anindya088,
Looks like you only need to replace the CSS selector
```css
.my-button.q-field__focusable-action {
    ...
}
```
with
```css
.my-button .q-field__focusable-action {
    ...
}
```

Without the space it means "an element with both classes "my-button" and "q-field__focusable-action".
With the space it means "an element with class "q-field__focusable-action" within an element with class "my-button".

**Reply by Anindya088** [reply]
Hi @falkoschindler , thanks a ton for this. Just now tried this and works beautifully.

I know probably these are very trivial questions, but usually for newbies like us, we keep reading all the docs and try out a bunch of things and sometimes get even more confused 😄 

Thank you for such specific and detailed explanations every time.

Regards,
Anindya
